git-cliff-core/src/changelog.rs

```diff
@@ -434,13 +433,44 @@ impl<'a> Changelog<'a> {
 				bitbucket_commits.clone(),
 				bitbucket_pull_request.clone(),
 			)?;
+		}
+
+		// render the header_template
+		if let Some(header_template) = &self.header_template {
+			let write_result = writeln!(
+				out,
+				"{}",
+				header_template.render(
+					&Releases {
+						releases: &releases,
+					},
+					Some(&additional_context),
+					&postprocessors,
+				)?
+			);
+			if let Err(e) = write_result {
+				if e.kind() != std::io::ErrorKind::BrokenPipe {
+					return Err(e.into());
+				}
+			}
+
+			// always insert an empty line after the header
+			if let Err(e) = out.write(b"\n") {
+				if e.kind() != std::io::ErrorKind::BrokenPipe {
+					return Err(e.into());
+				}
+			}
+		}
+
+		// render the body_template for each release
```

Comment:
unnecessary comment

Reply:
I disagree. `changelog.rs` is a huge file. The method generate especially is a long and very repetitive method, making it very difficult tor read unless you happen to have spent a lot of time in there. These simple comments help those who aren't familiar with the codebase to orient themselves.

Do you have a general policy as to what you would consider a necessary comment?

Reply:
There isn't a general policy but I usually try to avoid inline comments unless the code is in a unreadable / hard-to-understand state.

> These simple comments help those who aren't familiar with the codebase to orient themselves.

Yeah, you are right. But I think we should split the long functions into multiple parts instead and document each one. We can also add unit tests this way.